### PR TITLE
Fix exit code on service delete and revision delete

### LINKED ISSUE #960: "kn service delete" exited with code ZERO when it ran into failure
### Bug report

```
➜  ~ kn service delete hello
Internal error occurred: failed calling webhook "validation.webhook.serving.knative.dev": Post https://webhook.knative-serving.svc:443/resource-validation?timeout=10s: context deadline exceeded.
➜  ~ echo $?
0
➜  ~ kn service delete hello --no-wait
Internal error occurred: failed calling webhook "validation.webhook.serving.knative.dev": Post https://webhook.knative-serving.svc:443/resource-validation?timeout=10s: context deadline exceeded.
➜  ~ echo $?
0
➜  ~ kn service delete abc
services.serving.knative.dev "abc" not found.
➜  ~ echo $?
0
```

### Expected behavior

I suppose I should get a return code "1" for the delete failure.

### Steps to reproduce the problem

Delete n nonexistent ksvc:
```
➜  ~ kn service delete abc
services.serving.knative.dev "abc" not found.
➜  ~ echo $?
0
```

### kn version

```
➜  ~ kn version
Version:      v0.16.0
Build Date:   2020-07-15 07:01:17
Git Revision: 5cb78308
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.16.0)
* Eventing
  - sources.knative.dev/v1alpha2 (knative-eventing v0.16.0)
  - eventing.knative.dev/v1beta1 (knative-eventing v0.16.0)
```

### Knative (serving/eventing) version

> v0.16.0

### PR DESCRIPTION
**Pull request by hemanrnjn**
## Description

Fixed exit code issue on service delete

## Changes

* Fixed exit code to 1 which was earlier returned upon any failure of `kn service delete`

## Reference

Fixes #960

### CODE REVIEW

File: pkg/kn/commands/service/delete.go [hunk 1]
```diff
@@ -77,18 +78,22 @@ func NewServiceDeleteCommand(p *commands.KnParams) *cobra.Command {
...
+			errs := []string{}
 			for _, name := range args {
 				timeout := time.Duration(0)
 				if waitFlags.Wait {
 					timeout = time.Duration(waitFlags.TimeoutInSeconds) * time.Second
 				}
 				err = client.DeleteService(name, timeout)
 				if err != nil {
-					fmt.Fprintf(cmd.OutOrStdout(), "%s.\n", err)
+					errs = append(errs, err.Error())
```
Comment (maximilien) [inline pkg/kn/commands/service/delete.go:89]: I can comment this line and all tests still pass. This tells more there are no new tests for this change. Please add

**Comment by gosoon** [comment]
kn all command failed exited code is 0,does this need to be modified uniformly？

File: pkg/kn/commands/service/delete.go (hunk 1 above)
Comment (hemanrnjn) [reply pkg/kn/commands/service/delete.go:89]: @maximilien Added a test case. Now it should fail if the changes are reverted.

**Comment by hemanrnjn** [comment]
> kn all command failed exited code is 0,does this need to be modified uniformly？

Can you tell me more specifically with exactly what command you encountered this issue?

**Comment by navidshaikh** [comment]
/retest
  
      result_collector.go:75: ERROR: Error: ReconcileIngressFailed: Ingress reconciliation failed

**Comment by gosoon** [comment]
> > kn all command failed exited code is 0,does this need to be modified uniformly？
> 
> Can you tell me more specifically with exactly what command you encountered this issue?

I read the code and found no default error exit code，I tested some commands and found that part of the exit code was 1 and most of the exit code was 0.
```
$ kn service delete a
services.serving.knative.dev "a" not found.
$ echo $?
0
$ kn revision delete a
revisions.serving.knative.dev "a" not found.
$ echo $?
0
```

**Comment by navidshaikh** [comment]
+1 its the bug in revision delete command as well, @hemanrnjn might want to apply the fix for revisions too ?

**Comment by hemanrnjn** [comment]
> +1 its the bug in revision delete command as well, @hemanrnjn might want to apply the fix for revisions too ?

In this PR or a separate PR?

**Comment by maximilien** [comment]
LGTM once we can pass tests again

**Comment by hemanrnjn** [comment]
> LGTM once we can pass tests again. Also, should the "WIP" prefix be removed from title, or do you have more planned for this PR @hemanrnjn?

Not anything additional. Just looking into the cause of the failures of the integration tests as it might need some changes. I will remove WIP right after.

File: test/e2e/service_test.go [hunk 2]
```diff
@@ -136,12 +136,12 @@ func serviceMultipleDelete(r *test.KnRunResultCollector, existService, nonexistS
 	assert.Check(r.T(), !strings.Contains(out.Stdout, nonexistService), "The service", nonexistService, " exists (but is supposed to be not)")
 
 	out = r.KnTest().Kn().Run("service", "delete", existService, nonexistService)
-	r.AssertNoError(out)
```
Comment (navidshaikh) [inline test/e2e/service_test.go:139]: :+1:

**Pull request merged**